### PR TITLE
Resync fork on 2024-04 version

### PR DESCRIPTION
**Pull request by agateaux**
Resync with origin repository on April 2024.
Kept our own changes.

### CODE REVIEW

File: .github/auto-assignees.yaml
Comment (devinturner) [inline .github/auto-assignees.yaml:1]: Let's remove this file. We can readd codeowners at a later date.

File: .github/workflows/auto-assign-prs.yaml
Comment (devinturner) [inline .github/workflows/auto-assign-prs.yaml:1]: We can remove this file too.

File: MAINTAINERS
Comment (devinturner) [inline MAINTAINERS:1]: Can be removed from our fork

File: apis/goharbor.io/v1beta1/harborproject_types.go
Comment (devinturner) [inline apis/goharbor.io/v1beta1/harborproject_types.go:1]: @obanby look what we have here ❗ 

I don't think this is something to handle immediately, but now the buildstack can maintain "HarborProject" objects without having to know anything about the inner workings.

File: go.mod
Comment (devinturner) [inline go.mod:1]: This will downgrade a few dependencies. We should revert any of their downgrades.

**Review by devinturner — CHANGES_REQUESTED** [review]

File: apis/goharbor.io/v1beta1/harborproject_types.go
Comment (devinturner) [reply apis/goharbor.io/v1beta1/harborproject_types.go:1]: No change requested here, but just wanted to draw attention to it.

File: go.mod
Comment (agateaux) [reply go.mod:1]: I remember I got an issue with the upgrade. Some struct were not available any more in grpc, so the project didn't compile without a code refactoring....

File: .github/auto-assignees.yaml
Comment (agateaux) [reply .github/auto-assignees.yaml:1]: Done

File: .github/workflows/auto-assign-prs.yaml
Comment (agateaux) [reply .github/workflows/auto-assign-prs.yaml:1]: Done

File: MAINTAINERS
Comment (agateaux) [reply MAINTAINERS:1]: Done

File: go.mod
Comment (agateaux) [reply go.mod:1]: Done it worked.

**Review by devinturner — APPROVED** [review]
🚀 hell yes. Thank you for getting this ready.

**Comment by BRONSOLO** [comment]
@devinturner @agateaux is this needed for the 5.4-release?

**Comment by BRONSOLO** [comment]
Confirmed in 5.4 sync -- this is not needed for 5.4.

**Pull request merged**